sdk/eidolon_ai_sdk/util/stream_collector.py

```diff
@@ -62,11 +61,13 @@ async def __anext__(self):
         return event
 
 
-def stream_manager(stream: AsyncIterator[StreamEvent], context: StartStreamContextEvent):
+def stream_manager(
+    stream: AsyncIterator[StreamEvent] | Callable[[], AsyncIterator[StreamEvent]], context: StartStreamContextEvent
+):
     async def _iter():
         yield context.model_copy()
         try:
-            async for event in stream:
+            async for event in stream() if callable(stream) else stream:
```

Comment:
If we want to represent context error while creating the stream, it is nice to be able to pass in a factory to get the stream